packages/payload/src/utilities/createLocalReq.ts

```diff
@@ -56,7 +57,11 @@ export const createLocalReq: CreateLocalReq = (
   req.i18n = i18n
   req.t = i18n.t
   req.user = user || req?.user || null
-  if (!req.collection) req.collection = collection ? payload.collections?.[collection] : null
+  req.collection = collection
+    ? payload.collections?.[collection]
+      ? isolateObjectProperty(payload.collections[collection], collection)
+      : null
+    : null
```

Comment:
If we set `req.collection` to null when there is no collection, we get back the original problem:

```
 FAIL  test/_community/int.spec.ts (6.53 s)
  _Community Tests
    ✓ local API example (7 ms)
    ✕ rest API example (19 ms)

  ● _Community Tests › rest API example

    TypeError: Cannot read properties of undefined (reading 'text')

      68 |     }).then((res) => res.json())
      69 |
    > 70 |     expect(newPost.doc.text).toEqual('REST API EXAMPLE')
         |                        ^
      71 |   })
      72 | })
      73 |

      at Object.text (test/_community/int.spec.ts:70:24)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 passed, 2 total
Snapshots:   0 total
Time:        6.566 s, estimated 9 s
Ran all test suites matching /test\/_community\/int.spec.ts/i.
 ELIFECYCLE  Command failed with exit code 1.
```